examples/gno.land/r/demo/profile/profile.gno

```diff
@@ -0,0 +1,312 @@
+package profile
+
+import (
+	"bytes"
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/ufmt"
+	"std"
+	"time"
+)
+
+type UserProfile struct {
+	Username            string
+	Address             std.Address
+	AvatarURL           string
+	Age                 string
+	Gender              string
+	Website             string
+	Country             string
+	City                string
+	ModificationHistory map[string][]string
+}
+
+var profiles *avl.Tree
+var usernameIndex *avl.Tree // Index to quickly find a profile by username
+
+func init() {
+	profiles = avl.NewTree()
+	usernameIndex = avl.NewTree()
+}
+
+func CreateProfile(username, avatarurl, age, gender, website, country, city string) {
+	caller := std.GetOrigCaller()
+
+	// Check if a profile already exists for this address
+	if _, exists := profiles.Get(caller.String()); exists {
+		panic("A profile already exists for this address")
+	}
+
+	// Check if the username is already taken
+	if username != "" {
+		_, exists := usernameIndex.Get(username)
+		if exists {
+			panic("Username is already taken")
+		}
+	}
+
+	profile := UserProfile{
+		Username:  username,
+		Address:   caller,
+		AvatarURL: avatarurl,
+		Age:       age,
+		Gender:    gender,
+		Website:   website,
+		Country:   country,
+		City:      city,
+	}
+
+	profiles.Set(caller.String(), profile)
+
+	// Add the username to the index
+	if username != "" {
```

Comment:
This check will always pass at this point, because you have the same check above and panic if username is ""

Reply:
[commit](https://github.com/gnolang/gno/pull/1983/commits/38b5439a20dc4aa23b2970bd3f2cad2eebbf02bd)